### ISSUE TITLE
[feature request] [discussion] torch.lerp: optimize exact 0.0, exact 1.0 weight to force exact copy

### ISSUE THREAD

**Issue by vadimkantorov** [pytorch/pytorch#71701]
### 🚀 The feature, motivation and pitch

It may make sense for having exact copy behaviour for exact zero or one weights. Probably it's not a big issue for typical usecases of lerp, but sometimes it might bite some correctness. (although there may be performance repercussions if more checks are done...)

I looked at the CPU code: https://github.com/pytorch/pytorch/blob/af49805a73fb0a4b0459f25511598e5abfb54a96/aten/src/ATen/native/cpu/LerpKernel.cpp, it seems it does subtractions (end - start) in all code branches, it might be slightly inaccurate if they are small values because of floating point

**Comment by vadimkantorov** [comment]
This is even more true when called with scalar weights. Weight 0 should be no-op, weight 1 should be a copy

**Comment by ngimel** [comment]
Do you have an example of `lerp` producing inexact values? The way formulas are written there's a multiplication by `weight` or `weight-1` which would produce exact 0, and thus `lerp` would return exact `self` or `end` result.

**Comment by vadimkantorov** [comment]
I think you are right, that if-trick and using weight or 1-weight probably fixes it.

I was just looking at the code and was wondering if it does exact copies. I also wondered what's perf difference compared to regular copy (given the math and the if), but probably it's memory-speed bound anyway, right?

**Comment by ngimel** [comment]
Yes, it's bandwidth-bound, and weight of one is a corner case that's not specifically optimized for anyway. Closing for now, reopen if you think it's relevant.

**Comment by vadimkantorov** [comment]
@mgimel I guess the exact copy case is important where weight is 0, and the lhs is inf, then having an airthmetic formula might lead to nan if weight is multiplied by lhs

**Comment by vadimkantorov** [comment]
```
>>> torch.lerp(torch.tensor(float('inf')), torch.tensor(1.0), torch.tensor(1.0))
tensor(nan)
```

but the result should better be `tensor(1.0)`

**Comment by vadimkantorov** [comment]
@ngimel could you please reopen this issue? (as fixing this would also fix the masking problem where inf * 0 or nan * 0 gives unneccessarily nan outputs if the weights are actually meaning to mask out these bad values)
